Carcass, processing lot, and carcass observations resources

cookeac opened the pull request:
Add `icarCarcassResource`, `icarProcessingLot`, and `icarCarcassObservationsEvent` and supporting types and enumerations.

Resolves https://github.com/adewg/ICAR/issues/360
Resolves https://github.com/adewg/ICAR/issues/361
Resolves https://github.com/adewg/ICAR/issues/362
Resolves #363 
Resolves #155

cookeac committed:
Add organisations (identifiers, identity, type).

cookeac committed:
Extend consignment with org and declarations

Extend the icarConsignmentType adding origin and destination organisations, and additional information needed for sending livestock for processing.

cookeac committed:
Add carcasses, processing lots, and observations

Add `icarCarcassResource`, `icarProcessingLot`, and `icarCarcassObservationsEvent` and supporting types and enumerations.
Resolves #360
Resolves #361
Resolves #362

cookeac commented:
Hoping that Andreas can review this from a code perspective, and Erwin and Craig can look from a carcass data perspective.

AndreasSchultzGEA commented on resources/icarCarcassObservationsEventResource.json:
```diff
@@ -0,0 +1,41 @@
+                    }
+                },
+                "side": {
+                    "description": "The side of the carcass observed in this event ('Both` if not split).",
```
different quotation marks: 'Both`

cookeac replied:
```suggestion
                    "description": "The side of the carcass observed in this event (use Both if not split).",
```

AndreasSchultzGEA commented on types/icarCarcassObservationType.json:
```diff
@@ -0,0 +1,40 @@
+            "description": "The status of the observation.",
+            "$ref": "../enums/icarObservationStatusType.json"
+        },
+        "problemDetail": {
```
"remark" instead?

AndreasSchultzGEA commented on types/icarCarcassType.json:
```diff
@@ -0,0 +1,66 @@
+            "description": "The lot in which the carcass was processed. ",
+            "$ref": "../types/icarProcessingLotType.json"
+        },
+        "killDate": {
```
naming to killDateTime?
btw: killDateTime is also defined in ProcessingLot

cookeac replied:
```suggestion
        "killDateTime": {
```

AndreasSchultzGEA commented on types/icarCarcassType.json:
```diff
@@ -0,0 +1,66 @@
+            "description": "Plant-specific destination codes that are used for company requirements and to link to boning run processing.",
+            "type": "string"
+        },
+        "refProcessorGrid": {
```
just "processorGrid"?

AndreasSchultzGEA commented on types/icarOrganizationIdentityType.json:
```diff
@@ -0,0 +1,23 @@
+{
+    "description": "The identity of an organization in livestock supply chains. Based on a minimal set of identifiers from schema.org/organization.",
+    "type": "object",
+    "properties": {
```
no mandatory properties?

AndreasSchultzGEA commented on types/icarOrganizationType.json:
```diff
@@ -0,0 +1,36 @@
+                    "description": "The larger organization that this organization is a sub-organization of, if any.",
+                    "$ref": "../types/icarOrganizationIdentityType.json"
+                },
+                "membershipIdentifiers": {
```
what is the difference between "membershipIdentifiers" and "establishmentIdentifiers"?

cookeac replied:
**Establishment** identifiers are typically legal identifiers for an organisation or processing plant or farm. These may be the obvious (e.g. farm identifier) but may also be food safety or biosecurity identifiers (it would be great if there was just one, but there are often > 1).
**Membership** identifiers typically relate to a supply or assurance scheme - again, there may be multiple. These are optional - farms, locations, and processors can choose to be members of the scheme.
We could combine these into a single array, but then it is harder to distinguish them.

AndreasSchultzGEA commented on types/icarOrganizationType.json:
```diff
@@ -0,0 +1,36 @@
+        ,
+        {
+            "type": "object",
+            "properties": {
```
nothing required?

cookeac replied:
Resolved now - needs at least a name in icarOrganizationIdentityType

cookeac committed:
Correct quotation marks in descr.

cookeac committed:
Rename killDate to killDateTime for consistency

cookeac commented on types/icarCarcassType.json:
```diff
@@ -0,0 +1,66 @@
+            "description": "Plant-specific destination codes that are used for company requirements and to link to boning run processing.",
+            "type": "string"
+        },
+        "refProcessorGrid": {
```
```suggestion
        "processorGrid": {
```

cookeac committed:
simplify processorGrid name

cookeac commented on types/icarOrganizationIdentityType.json:
```diff
@@ -0,0 +1,23 @@
+{
+    "description": "The identity of an organization in livestock supply chains. Based on a minimal set of identifiers from schema.org/organization.",
+    "type": "object",
```
```suggestion
    "type": "object",
    "required": ["name"],
```

cookeac committed:
Make organization name required.

cookeac committed:
Fix Spectral linting errors in repro URL scheme

cookeac committed:
Change problemDetail to remark for consistency

cookeac committed:
Rename qualitativeGrade to qualitativeValue

I've had feedback that grade is confusing if the text value is not specifically a grade but some other result (e.g. the presence of a defect).
